src/utils/exec-promise.ts

```diff
@@ -1,14 +1,52 @@
-import * as childProcess from 'child_process';
-import { promisify } from 'util';
+import { spawn } from 'child_process';
 
-const exec = promisify(childProcess.exec);
+/**
+ * Wraps up running a command into a single promise,
+ * returning the stdout as a string if the command succeeds
+ * and throwing if it does not.
+ *
+ * @param cmd the command as a string to pass in
+ */
+export default async function execPromise(cmd: string, args?: string[]) {
+  return new Promise<string>((completed, reject) => {
```

Comment:
May I recommend https://github.com/sindresorhus/execa for this sort of thing? We don't have to handle the logic and can rely on a good open source library to get it right.

Reply:
Yeah I'm down with that

Reply:
but that can be a separate PR. Just get the tests passing on this one + rewrite the commits to @orta and I'll merge